ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -361,40 +269,51 @@ public class ResourceDownloadManager {
                                    isUpdate: Bool,
                                    fetchRepositoryIfNeeded: Bool = true,
                                    completionBlock: CompletionHandler<InstallableLexicalModel>? = nil) {
-    
-    // TODO:  We should always force a refetch after new keyboards are installed so we can redo our language queries.
-    //        That should probably be done on successful keyboard installs, not here, though.
-    if fetchRepositoryIfNeeded {
-      // A temp measure to make sure things aren't totally broken.  Definitely not optimal.
-      Manager.shared.apiLexicalModelRepository.fetch(completionHandler: nil)
-    }
-    
-    guard let _ = Manager.shared.apiLexicalModelRepository.lexicalModels else {
-      if fetchRepositoryIfNeeded {
-        log.info("Fetching repository from API for lexicalModel download")
-        Manager.shared.apiLexicalModelRepository.fetch(completionHandler: fetchHandler(for: .lexicalModel) {
-          self.downloadLexicalModel(withID: lexicalModelID, languageID: languageID, isUpdate: isUpdate, fetchRepositoryIfNeeded: false, completionBlock: completionBlock)
-        })
+    // Note:  in this case, someone knows the "full ID" of the model already, but NOT its location.
+    //        For lexical models, we can use either the LexicalModel query or the PackageVersion query.
+    //        For consistency with keyboard download behavior, we use the PackageVersion query here.
```

Comment:
Yes, I could have used the lexical-model search query for this... but that won't be an option for keyboards, which will need similar support for their legacy API.

Reply:
On the plus side, this gets me that much closer to replacing both lexical model and keyboard updates, as this provided the first use-case for integration of that code path.